src/tests/multihost/alltests/test_krb_access_provider.py

```diff
@@ -0,0 +1,177 @@
+
+        lookup_u = "auth_success testuser3 Secret123"
+        cmd = multihost.client[0].run_command(lookup_u)
+        assert cmd.returncode == 0
+
+        time.sleep(30)
+
+        strings = ["pam_sss(sshd:auth): authentication success", "Accepted password for testuser3", "Access allowed for user \[testuser3"]
+        logfiles = ['/var/log/secure', "/var/log/secure", '/var/log/sssd/sssd_%s.log' % ds_instance_name]
+
+        for index, string in enumerate(strings):
+            logfile = logfiles[index]
+            check_pattern_in_file(string, logfile, multihost)
+
+        path = "/home/testuser3/.k5login"
+        erase_file(path, multihost)
+
+    def test_0003_k5login_user3_with_user4(multihost):
+        """
+        """
+        paths = ["/var/log/secure", '/var/log/sssd/sssd_%s.log' % ds_instance_name]
+        for path in paths:
+            erase_file(path, multihost)
+
+        echo_cmds = ['echo "testuser3@EXAMPLE.COM" > /home/testuser3/.k5login', 'echo "testuser4@EXAMPLE.COM" > /home/testuser3/.k5login']
+        for command in echo_cmds:
+            multihost.client[0].run_command(command)
+
+        restore_selinux = "restorecon -v /home/testuser3/.k5login"
+        multihost.client[0].run_command(restore_selinux)
+
+        lookup_u = "auth_success testuser3 Secret123"
+        cmd = multihost.client[0].run_command(lookup_u)
+        assert cmd.returncode == 0
+
+        # WRITE_CODE_HERE
+
+        string = "testuser3"
+        logfile = "/tmp/accessProvider_id_krb5_003.out"
+        check_pattern_in_file(string, logfile, multihost)
+
+        path = "/home/testuser3/.k5login"
+        erase_file(path, multihost)
+
+    def test_0004_k5login_user4_with_deleted_user3(multihost):
+        """
+        """
+        paths = ["/var/log/secure", '/var/log/sssd/sssd_%s.log' % ds_instance_name]
+        for path in paths:
+            erase_file(path, multihost)
+
+        echo_cmds = ['echo "testuser3@EXAMPLE.COM" > /home/testuser3/.k5login', 'echo "testuser4@EXAMPLE.COM" > /home/testuser3/.k5login']
+        for command in echo_cmds:
+            multihost.client[0].run_command(command)
+
+        restore_selinux = "restorecon -v /home/testuser3/.k5login"
+        multihost.client[0].run_command(restore_selinux)
+
+        lookup_u = "auth_success testuser3 Secret123"
+        cmd = multihost.client[0].run_command(lookup_u)
+        assert cmd.returncode == 0
+
+        # WRITE_CODE_HERE_1
+
+        time.sleep(5)
+
+        # WRITE_CODE_HERE_2
+
+        string = "testuser4"
+        logfile = "/tmp/accessProvider_id_krb5_004.out"
+        check_pattern_not_in_file(string, logfile, multihost)
+
+        string = "testuser4"
+        logfile = "/tmp/accessProvider_id_krb5_004.out"
+        check_pattern_in_file(string, logfile, multihost)
+
+
+def erase_file(path, multihost):
+    cmd = "rm -rf %s" % path
+    multihost.client[0].run_command(cmd)
+    multihost.client[0].run_command("touch %s", path)
```

Comment:
Please don't use this.
It is wrong on so many levels:
1. Recursive delete should not be used if You are deleting just a file.
2. erase_file is misleading as it tries to erase file on client[0] and will not work for master.
3. passed multihost is the whole fixture instead of having parameter like machine that will contain the intended host like multihost.client[0]
4. There is no sanity check for the provided path.
5. You are creating empty file instead of deleted old one, which is unexpected behavior.
6. To replace file with an empty one the truncate is the right command.
7. The newly created empty file will have different ownership and possibly also selinux content that the original one.
8. The function is without any meaningful docstring.

If You are deleting file just call directly the the multihost.client[0].run_command("rm -f myfile") for best readability.